### PR TITLE
Adding annotations support for statefulset-read.yaml

### LINKED ISSUE #11688: No annotations for helm-loki chart read StatefulSet
**Describe the bug**
There is no `annotations` argument for `helm-loki` chart in the read chart for `StatefulSet` kind yaml file.
In `values.yaml` file there is an argument for the `annotation`, but without any use.

**To Reproduce**
1. Read `helm-loki` -> `templates` -> `read` -> `statefulset-read.yaml` file.

**Expected behavior**
The same behavior as the `write` chart `StatefulSet` kind yaml file, a merge of the "global" annotations, and the specific annotations for the relevant component in the values.yaml file.

**Environment:**
 - Infrastructure: `Kubernetes`
 - Deployment tool: `helm`

**Screenshots, Promtail config, or terminal output**
`statefulset-read.yaml` file: (missing the annotations use)

```
/apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: {{ include "loki.readFullname" . }}
  namespace: {{ $.Release.Namespace }}
  labels:
    app.kubernetes.io/part-of: memberlist
    {{- include "loki.readLabels" . | nindent 4 }}
spec:
...

```

On the other hand, this is how it's handled for `write's` `statefulset-write.yaml` file: (the proposal solution)

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: {{ include "loki.writeFullname" . }}
  namespace: {{ $.Release.Namespace }}
  labels:
    {{- include "loki.writeLabels" . | nindent 4 }}
    app.kubernetes.io/part-of: memberlist
  {{- if or (not (empty .Values.loki.annotations)) (not (empty .Values.backend.annotations))}}
  annotations:
    {{- with .Values.loki.annotations }}
    {{- toYaml . | nindent 4 }}
    {{- end }}
    {{- with .Values.write.annotations }}
    {{- toYaml . | nindent 4 }}
    {{- end }}
  {{- end }}
spec:
...

```

### PR DESCRIPTION
**Pull request by YogevLevy93**
Adding fixing lake of annotations support for `read` StatefulSet

**What this PR does / why we need it**:
Currently there is no use for `read.annotations` at the `statefulset-read.yaml` file.
The key exists on `values.yaml` file, with no use.

**Which issue(s) this PR fixes**:
Fixes #11688 

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by JStickler** [comment]
Please note this bullet point from the PR checklist:
For Helm chart changes bump the Helm chart version in production/helm/loki/Chart.yaml and update production/helm/loki/CHANGELOG.md and production/helm/loki/README.md. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

File: production/helm/loki/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,9 @@ Entries should include a reference to the pull request that introduced the chang
 
 [//]: # (<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line.)
 
+## 5.41.8
```
Comment (MichelHollands) [inline production/helm/loki/CHANGELOG.md:16]: Can you make this 5.41.9?

Also you checked this but did not do the steps in it. For Helm chart changes bump the Helm chart version in production/helm/loki/Chart.yaml and update production/helm/loki/CHANGELOG.md and production/helm/loki/README.md. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

Comment (MichelHollands) [reply production/helm/loki/CHANGELOG.md:16]: I merged in main, which might have caused some of those changes to disappear as they are the same as main is now.

Comment (YogevLevy93) [reply production/helm/loki/CHANGELOG.md]: @MichelHollands  Done

**Review by MichelHollands — APPROVED** [review]

**Pull request merged**